src/main/scala/io/citrine/lolo/validation/Metric.scala

```diff
@@ -0,0 +1,212 @@
+    val model = Math.sqrt(2 / (Math.PI - 2)) * Math.sqrt(sigmaVar) / sigmaMean
+    val estimate = computeFromPredictedUncertaintyActual(baseline)
+    // println(model, estimate)
+    */
+
+    computeFromPredictedUncertaintyActual(predictedUncertaintyActual) / computeFromPredictedUncertaintyActual(ideal)
+  }
+
+  /**
+    * Covariance(X, Y) / Sqrt(Var(X) * Var(Y)), where X is predicted uncertainty and Y is magnitude of error
+    */
+  def computeFromPredictedUncertaintyActual(
+                                             pua: Seq[(Double, Double, Double)]
+                                           ): Double = {
+    val error = pua.map{case (p, _, a) => Math.abs(p - a)}
+    val sigma = pua.map(_._2)
+
+    val meanError = error.sum / error.size
+    val varError = error.map(x => Math.pow(x - meanError, 2.0)).sum / error.size
+    val meanSigma = sigma.sum / sigma.size
+    val varSigma = sigma.map(x => Math.pow(x - meanSigma, 2.0)).sum / sigma.size
+
+    val covar = error.zip(sigma).map{case (x, y) => (x - meanError) * (y - meanSigma)}.sum / sigma.size
+    covar / Math.sqrt(varError * varSigma)
+  }
+}
+
+object Metric {
+
+  /**
+    * Estimate a set of named metrics by applying them to multiple sets of predictions and actual values
+    *
+    * The uncertainty in the estimate of each metric is calculated by looking at the variance across the batches
+    *
+    * @param pva predicted-vs-actual data in a series of batches
+    * @param metrics to apply to the predicted-vs-actual data
+    * @return map from the metric name to its (value, uncertainty)
+    */
+  def estimateMetrics[T](
+                          pva: Iterable[(PredictionResult[T], Seq[T])],
+                          metrics: Map[String, Metric[T]]
+                        ): Map[String, (Double, Double)] = {
+
+    pva.flatMap{ case (predictions, actual) =>
+      // apply all the metrics to the batch at the same time so the batch can fall out of memory
+      metrics.mapValues(f => f.evaluate(predictions, actual)).toSeq
+    }.groupBy(_._1).mapValues{x =>
+      val metricResults = x.map(_._2)
+      val mean = metricResults.sum / metricResults.size
+      val variance = metricResults.map(y => Math.pow(y - mean, 2)).sum / metricResults.size
+      (mean, Math.sqrt(variance / metricResults.size))
+    }
+  }
+
+  /**
+    * Compute metrics as a function of a parameter, given a builder that takes the parameter to predicted-vs-actual data
+    * @param parameterName name of the parameter that's being scanned over
+    * @param parameterValues values of the parameter to try
+    * @param metrics to apply at each parameter value
+    * @param logScale whether the parameters should be plotted on a log scale
+    * @param pvaBuilder function that takes the parameter to predicted-vs-actual data
+    * @return an [[XYChart]] that plots the metrics vs the parameter value
+    */
+  def scanMetrics[T](
```

Comment:
It was not intuitive that this would produce a chart. I would've expected `scanMetrics` to generate results (of some kind). Then I would expect to use `visualize` to generate the plot. I'm not sure it would be better with the proposed way (since you ultimately want the plot). Just flagging it as something that seemed odd.

Reply:
Good point; renamed.